statistics/views.py

```diff
@@ -79,44 +79,102 @@ def get_state_info(request, code):
 
 def training(request):
     """ Organiser index page """
-    collectionSet = None
-    state = None
-
     collectionSet = TrainingRequest.objects.filter(
-        participants__gt=0,
-        sem_start_date__lte=datetime.now()
-    ).order_by('-sem_start_date')
-    header = {
-        1: SortableHeader('#', False),
-        2: SortableHeader('training_planner__academic__state__name', True, 'State'),
-        3: SortableHeader('training_planner__academic__city__name', True, 'City'),
-        4: SortableHeader('training_planner__academic__institution_name', True, 'Institution'),
-        5: SortableHeader('course__foss__foss', True, 'FOSS'),
-        6: SortableHeader('department', True, 'Department'),
-        7: SortableHeader('course__category', True, 'Type'),
-        8: SortableHeader('training_planner__organiser__user__first_name', True, 'Organiser'),
-        9: SortableHeader('sem_start_date', True, 'Date'),
-        10: SortableHeader('participants', 'True', 'Participants'),
-        11: SortableHeader('Action', False)
-    }
+            sem_start_date__lte=datetime.now()
+        ).order_by('-sem_start_date')
+    state = None
+    TRAINING_COMPLETED = '1'
+    TRAINING_PENDING = '0'
+
+    if request.method == 'GET':
+        status = request.GET.get('status')
+        if status not in [TRAINING_COMPLETED, TRAINING_PENDING]:
+            status = TRAINING_COMPLETED
+
+    if status == TRAINING_PENDING:
+        collectionSet = collectionSet.filter(participants=0)
+        header = {
+            1: SortableHeader('#', False),
+            2: SortableHeader('training_planner__academic__state__name', True, 'State'),
+            3: SortableHeader('training_planner__academic__city__name', True, 'City'),
+            4: SortableHeader('training_planner__academic__institution_name', True, 'Institution'),
+            5: SortableHeader('course__foss__foss', True, 'FOSS'),
+            6: SortableHeader('department', True, 'Department'),
+            7: SortableHeader('course__category', True, 'Type'),
+            8: SortableHeader('training_planner__organiser__user__first_name', True, 'Organiser'),
+            9: SortableHeader('sem_start_date', True, 'Date'),
+            10: SortableHeader('participants', 'True', 'Participants'),
+            11: SortableHeader('Action', False)
+        }
+    else:
+        collectionSet = collectionSet.filter(participants__gt=0)
```

Comment:
Since collectionSet is the only thing that changes based on `if`, let's just put the `collectionSet` in the if condition and do not duplicate the `header` variable.

```python
if status == TRAINING_PENDING:
  collectionSet = collectionSet.filter(participants=0)
else:
  collectionSet = collectionSet.filter(participants__gt=0)

header = {
   # headers definition here.
}
```